app/README.md

```diff
@@ -52,9 +52,127 @@ committing changes, otherwise the tests will be skipped on CI.
-into code that forms a design system. The design system contains all design
-primitives which can be composed to create rich user experiences.
+being improved and extended. This style guide is translated into code that
+forms a design system. The design system contains all design primitives which
+can be composed to create rich user experiences.
+
+Most of the components defined by the design system can be conveniently seen on
+one page within the app UI by pressing <kbd>shift</kbd> + <kbd>D</kbd>. This
+will bring up the Design System Guide modal. To close the modal, hit
+<kbd>ESC</kbd>.
+
+The purpose of the Guide is to be able to see how changes to components affect
+all variations in one glance. Newly created components should be added to the
+Guide, explaining all the different variations and use cases a component can be
+used.
+
+
+#### File structure
+
+To be able to find code and make changes quickly we organize our components in
+groups defined by use-case, re-usability and complexity. These groups are:
+
+  - `DesignSystem/Primitives`: simple, yet highly reusable components like
+    typography, buttons, form elements, spacing, positioning and other
+    utilities.
+
+    Components in this group are usually just wrappers around standard HTML
+    elements with custom styling.
+
+    There are currently two ways of organizing primitives:
+
+      - all-in-one components where the type of the component is passed down
+        via a `variant` prop. This is for components which have a very
+        similar markup, but whose styling differs across variants.
+        Examples in this category are: buttons, typography and positioning
+        helpers.
+
+      - namespaced components, where the markup is very different across
+        variants, like: form elements and icons.
+
+    To figure which way to write a new primitive component, start by looking at
+    how it's going to be used in the code and then go for the better
+    ergonomics.
+
+    All public primitives are exported via a central `index.js` file, which
+    makes usage as easy as:
+
+    ```
+    <script>
+      import { Button, Text, Icon, Input } from "../DesignSystem/Primitives";
+    </script>
+
+    <Icon.Home />
+    <Button variant="secondary">OK</Button>
+    <Text variant="hugeTitle">Radicle</Text>
+    ```
+
+
+  - `DesignSystem/Components`: reusable low-to-high complexity components,
+    layouts and single-use fragments which are extracted from screens or other
+    components for the sake of code readability.
+
+    Sub-folders in this category should only be created for breaking up larger
+    components into smaller fragments. If a component is broken up in
+    fragments, make sure to only export the component for public use, not the
+    fragments.
+
+    Single-use components that are extracted from screens and used by only one
+    screen should be prefixed with the name of the screen, for example:
+    `DesignSystemGuideSection.svelte`.
```

Comment:
If it's only relevant for one screen, what's the benefit of moving it into the design system in the first place?

Reply:
See below about partials.